packages/core/src/RenderingEngine/WSIViewport.ts

```diff
@@ -233,29 +253,41 @@ class WSIViewport extends Viewport {
         Modality: this.modality,
         FrameOfReferenceUID: this.frameOfReferenceUID,
       },
-      imageData: {
-        getDirection: () => metadata.direction,
-        getDimensions: () => metadata.dimensions,
-        getRange: () => [0, 255],
-        getScalarData: () => this.getScalarData(),
-        getSpacing: () => metadata.spacing,
-        worldToIndex: (point: Point3) => {
-          const canvasPoint = this.worldToCanvas(point);
-          const pixelCoord = this.canvasToIndex(canvasPoint);
-          return [pixelCoord[0], pixelCoord[1], 0] as Point3;
-        },
-        indexToWorld: (point: Point3) => {
-          const canvasPoint = this.indexToCanvas([point[0], point[1]]);
-          return this.canvasToWorld(canvasPoint);
-        },
-      },
+
       hasPixelSpacing: this.hasPixelSpacing,
       calibration: this.calibration,
       preScale: {
         scaled: false,
       },
       scalarData: this.getScalarData(),
+      imageData,
+      voxelManager: {
```

Comment:
I don't quite get this - it doesn't have the rest of hte voxel manager functions.  Can you at least comment what this is doing?

Reply:
It is for the annotations to work, since all of them work on voxelManager and not on scalarData now, i added a comment